hack/generator/pkg/codegen/pipeline_create_arm_types.go

```diff
@@ -196,22 +192,25 @@ func transformTypeDefinition(
 	def astmodel.TypeDefinition,
 	handlers []conversionHandler) (astmodel.TypeDefinition, error) {
 
-	originalType, ok := def.Type().(*astmodel.ObjectType)
-	if !ok {
-		return astmodel.TypeDefinition{}, errors.Errorf("input type %q (%T) was not of expected type Object", def.Name(), def.Type())
+	visitor := astmodel.MakeTypeVisitor()
+	visitor.VisitObjectType = func(this *astmodel.TypeVisitor, it *astmodel.ObjectType, ctx interface{}) (astmodel.Type, error) {
```

Comment:
Should this keep track of whether it actually found any object type in the definition? The old version of the code would return an error if called on a non-object type definition but I think this would quietly succeed without applying the handlers. (Which might be fine, not sure!)

Reply:
It should error if no change is made; I'll add in a check - thanks for catching this.